### PR TITLE
Resolve ignored code check

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Breaking change: no

The given code violates some of our own rules and I'm not even sure how it should be evaluated :D

### CODE REVIEW

File: src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php [hunk 1]
```diff
@@ -5742,10 +5742,15 @@ protected function parseFormalParameterOrTypeHintOrByReference()
...
         $this->tokenStack->push();
 
-        switch ($tokenType) {
-            case $this->isTypeHint($tokenType) && ($typeHint = $this->parseOptionalTypeHint()):
+        if ($this->isTypeHint($tokenType)) {
+            $typeHint = $this->parseOptionalTypeHint();
+            if ($typeHint) {
                 $parameter = $this->parseFormalParameterAndTypeHint($typeHint);
-                break;
+                return $this->setNodePositionsAndReturn($parameter);
```
Comment (AJenbo) [inline src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:5749]: This line could be avoided if the block is moved to the switch default, but I'm unsure if it's ok to test this last rather the first.

Comment (kylekatarnls) [reply src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:5749]: I suggest to decouple in 2 methods where a first one would get the proper parameter according to type, then the `parseFormalParameterOrTypeHintOrByReference()` would just call it and pass it to `setNodePositionsAndReturn()` this way, it's more clear what is conditional (the parameter calculation) and what is not (the `setNodePositionsAndReturn`)

Comment (AJenbo) [reply src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php]: Done, see if you like the naming et al

**Review by kylekatarnls — APPROVED** [review]

File: src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php (hunk 1 above)
Comment (kylekatarnls) [reply src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php]: 👌

**Pull request merged**